src/utils/validateIdentifierInputs.ts

```diff
@@ -69,6 +78,7 @@ export const validateIdentifierInputs = (
       try {
         isValidPaymentQR(value);
       } catch (e) {
+        e.message = invalidIdentifierInputErr;
```

Comment:
for this case, will it be other errors? will it be safe to assume its a `invalidIdentifierInputErr`?

Reply:
isValidPaymentQR throw all the errors using the same error message: ERROR_MESSAGE.INVALID_IDENTIFIER_INPUT. However, the error is defined by a specific Error class (e.g. PaymentQRUnsupportedError, etc.), and CustomerQuotaScreen would check the type of error by instanceof the class. 

```
        case cartError instanceof PaymentQRDeformedError ||
          cartError instanceof PaymentQRMissingInfoError ||
          cartError instanceof PaymentQRUnsupportedError:
          showErrorAlert(cartError, () => clearCartError());
          return;
```

In alert.tsx, getTranslationKeyFromError will map the err.name (defined in the class) to the key in i18n using errorNameToTranslationKeyMappings instead of getTranslationKeyFromErrorMessage to display the correct title and body.

```
const errorNameToTranslationKeyMappings: Record<string, string> = {
  .....
  ........
  .......
  PaymentQRDeformedError: "deformedPaymentQR",
  PaymentQRMissingInfoError: "missingInfoInPaymentQR",
  PaymentQRUnsupportedError: "unsupportedPaymentMethodInPaymentQR",
};
```

In order to differentiate between whether the text is disabled, the message need to be replaced by invalidIdentifierInputErr. Finally, this allow showErrorAlert to check if the text is disabled using the error message to remove "Enter or" and change first letter to upper case without introducing additional messages in i18n.

```
....
.....
      showAlert({
        alertType: "ERROR",
        title: i18n.t(`errorMessages.${translationKey}.title`) ?? "Error",
        description:
          error.message ===
            ERROR_MESSAGE.INVALID_IDENTIFIER_INPUT_TEXT_DISABLED ||
          error.message === ERROR_MESSAGE.MISSING_IDENTIFIER_INPUT_TEXT_DISABLED
            ? errDescription
                .replace("Enter or ", "")
                .replace(/(^\w)/g, (m) => m.toUpperCase())
            : errDescription,
...
.......
```

Reply:
understand, thanks for the super detailed explanation!